packages/astro/test/astro-css-bundling.test.js

```diff
@@ -19,42 +19,91 @@ const UNEXPECTED_CSS = [
+					const link = $(`link[rel="stylesheet"][href^="${href}"]`);
+					expect(link.length).to.be.greaterThanOrEqual(1);
+					const outHref = link.attr('href');
+					builtCSS.add(outHref.startsWith('../') ? outHref.slice(2) : outHref);
+				}
+	
+				// test 2: assert old CSS was removed
+				for (const href of UNEXPECTED_CSS) {
+					const link = $(`link[rel="stylesheet"][href="${href}"]`);
+					expect(link).to.have.lengthOf(0);
+				}
+	
+				// test 3: assert all bundled CSS was built and contains CSS
+				for (const url of builtCSS.keys()) {
+					const bundledCss = await fixture.readFile(url);
+					expect(bundledCss).to.be.ok;
+				}
+			}
+		});
+
+		it('there are 4 css files', async () => {
+			const dir = await fixture.readdir('/assets');
+			expect(dir).to.have.a.lengthOf(4);
+		});
+
+		it('CSS includes hashes', async () => {
+			const [firstFound] = await fixture.readdir('/assets');
+			expect(firstFound).to.match(/[a-z]+\.[0-9a-z]{8}\.css/);
 		});
-		await fixture.build({ mode: 'production' });
 	});
 
-	it('Bundles CSS', async () => {
-		const builtCSS = new Set();
+	describe('using custom assetFileNames config', () => {
+		before(async () => {
+			fixture = await loadFixture({
+				root: './fixtures/astro-css-bundling/',
 
-		// for all HTML files…
-		for (const [filepath, css] of Object.entries(EXPECTED_CSS)) {
-			const html = await fixture.readFile(filepath);
-			const $ = cheerio.load(html);
+				vite: {
+					build: {
+						rollupOptions: {
+							output: {
+								assetFileNames: "assets/[name][extname]",
+								entryFileNames: "[name].js",
+							}
+						}
+					}
+				}
+			});
+			await fixture.build({ mode: 'production' });
+		});
 
-			// test 1: assert new bundled CSS is present
-			for (const href of css) {
-				const link = $(`link[rel="stylesheet"][href^="${href}"]`);
-				expect(link.length).to.be.greaterThanOrEqual(1);
-				const outHref = link.attr('href');
-				builtCSS.add(outHref.startsWith('../') ? outHref.slice(2) : outHref);
-			}
+		it('there are 4 css files', async () => {
+			const dir = await fixture.readdir('/assets');
+			expect(dir).to.have.a.lengthOf(4);
+		});
 
-			// test 2: assert old CSS was removed
-			for (const href of UNEXPECTED_CSS) {
-				const link = $(`link[rel="stylesheet"][href="${href}"]`);
-				expect(link).to.have.lengthOf(0);
-			}
+		it('CSS does not include hashes hashes', async () => {
+			const [firstFound] = await fixture.readdir('/assets');
+			expect(firstFound).to.not.match(/[a-z]+\.[0-9a-z]{8}\.css/);
+		});
 
-			// test 3: assert all bundled CSS was built and contains CSS
-			for (const url of builtCSS.keys()) {
-				const bundledCss = await fixture.readFile(url);
-				expect(bundledCss).to.be.ok;
-			}
-		}
+		it('there are 2 index named CSS files', async () => {
+			const dir = await fixture.readdir('/assets');
+			const indexNamedFiles = dir.filter(name => name.startsWith('index'))
+			expect(indexNamedFiles).to.have.a.lengthOf(2);
```

Comment:
Should this also test that the file names are not identical?

Reply:
These are the actual files written to disk. You can't have 2 files of the same name in the same folder. So they can't be the same.